lib/mix/lib/mix/tasks/test.ex

```diff
@@ -327,6 +327,16 @@ defmodule Mix.Tasks.Test do
   def run(args) do
     {opts, files} = OptionParser.parse!(args, strict: @switches)
 
+    files =
+      if Mix.Task.recursing?() do
+        case Regex.run(~r|.*\/apps\/(.*?)$|, File.cwd!()) do
```

Comment:
This looks promising! Unfortunately matching on `/apps/` could led to false positives. For example, someone may have all `apps` in an `apps` directory.

The good news is, if we are recursing, then we must be inside an umbrella, and we can get the current app with `Mix.Project.config[:app]`. So here is what I would do:

1. Check if we are recursing
2. If so, get the current app name
3. Then traverse test files. If the test file starts with "apps/", then we check if it belongs to "apps/#{current_app}", if so, we keep it without the "apps/#{current_app}" prefix, otherwise we discard it (since it belongs to another app than the current one)

Thoughts?

Reply:
Yes, thanks, that was my initial approach, but I blanked on how to get the current app name in a task. Pushed a new version with this approach.